devito/passes/iet/parpragma.py

```diff
@@ -284,6 +285,10 @@ def _select_candidates(self, candidates):
                 if i.is_Vectorized:
                     break
 
+                # Also, we do not want to collapse small atomic reductions
+                if i.is_ParallelAtomic and i.dim.is_Custom:
```

Comment:
It "small" but not amount of work small. it's more that collapsing atomic reduction over Custom dimension is very unlikely to improve performance as it will only increase race conditions

Reply:
CustomDimensions can be anything, they might even be equivalent in all to SpaceDimensions

What kind of CustomDimensions do you have in mind here? What loops are we trying to avoid to collapse?

Reply:
Just trying to prevent 
```
for (p_src ...)
 px = 
 py = 
 # omp for collapse
 for (rx = ...)
    for (ry = ...)
```

I could maybe make a special dims for this

Reply:
> It "small" but not amount of work small 

These loops should be pretty small, and the way "work" is defined later is essentially "number of iterations"

So I'm not sure I follow? These CustomDimensions have numeric size (right? or is this what I'm getting wrong?) so it should be feasible?